### PR TITLE
PR2: attributes rename

### PR DESCRIPTION
**Pull request by Saksham20**
There are a fair bit of change in this PR, will it be better to break it into 2 parts? 

Here are the changes: 
1. Images attributes: `self._image_raw`, `self._image_correlation`, these are subsequently packed into a dict using `self.get_images_dict()`
2. Traces attributes: `self._roi_response_raw`, `self._roi_response_dff`,  `self._roi_response_neuropil,` `self._roi_response_deconvolved`, these are then packed into dict using `self.get_traces_dict()`
3. all the property methods are removed from segmentationextractor base class. 
4. Renaming for more clarity: `filepath` > `file_path,` `no_of_channels` to `num_of_channels`

### CODE REVIEW

File: roiextractors/extractors/nwbextractors/nwbextractors.py [hunk 1]
```diff
@@ -252,16 +253,21 @@ def __init__(self, filepath):
             self.property_vals.append(np.array(ps[i].data))
 
         #Extracting stores images as GrayscaleImages:
-        self._greyscaleimages = [_nwbchildren_name[f] for f, u in enumerate(_nwbchildren_type) if u == 'GrayscaleImage']
+        _greyscaleimages = [i for i in nwbfile.all_children() if type(i).__name__ == 'GrayscaleImage']
+        self._image_correlation = [i.data[()] for i in _greyscaleimages if 'corr' in i.name.lower()][0]
+        self._image_mean = [i.data[()] for i in _greyscaleimages if 'mean' in i.name.lower()][0]
```
Comment (bendichter) [inline roiextractors/extractors/nwbextractors/nwbextractors.py:258]: I feel like there is probably a better way to do this, but we can address this later

File: roiextractors/extractors/schnitzerextractor/cnmfesegmentationextractor.py [hunk 2]
```diff
@@ -55,23 +52,23 @@ def _tot_exptime_extractor_read(self):
         return self._dataset_file[self._group0[0]]['time']['totalTime'][0][0]
 
     def _summary_image_read(self):
-        summary_images_ = self._dataset_file[self._group0[0]]['Cn']
-        return np.array(summary_images_).T
+        summary_image = self._dataset_file[self._group0[0]]['Cn']
+        return np.array(summary_image).T
```
Comment (bendichter) [inline roiextractors/extractors/schnitzerextractor/cnmfesegmentationextractor.py:56]: shouldn't this be the correlation image?

File: roiextractors/extractors/suite2p/suite2psegmentationextractor.py [hunk 3]
```diff
@@ -53,6 +51,17 @@ def get_accepted_list(self):
...
+    def _summary_image_read(self, bstr='meanImg'):
+        img = None
+        if bstr in self.ops:
+            if bstr == 'Vcorr' or bstr == 'max_proj':
+                img = np.zeros((self.ops['Ly'], self.ops['Lx']), np.float32)
+                img[self.ops['yrange'][0]:self.ops['yrange'][-1],
+                self.ops['xrange'][0]:self.ops['xrange'][-1]] = self.ops[bstr]
+            else:
+                img = self.ops[bstr]
+        return img
```
Comment (bendichter) [inline roiextractors/extractors/suite2p/suite2psegmentationextractor.py:63]: I don't really understand what's going on here

File: roiextractors/segmentationextractor.py [hunk 4]
```diff
@@ -215,25 +154,66 @@ def get_image_size(self) -> ArrayType:
...
+    def get_traces_dict(self):
+        """
+        Returns traces as a dictionary with key as the name of the ROiResponseSeries
+        Returns
+        -------
+        _roi_response_dict: dict
+            dictionary with key, values representing different types of RoiResponseSeries
+            Flourescence, Neuropil, Deconvolved, Background etc
+        """
+        return deepcopy(dict(raw=np.array(self._roi_response_raw),
```
Comment (bendichter) [inline roiextractors/segmentationextractor.py:185]: I don't think you need a `deepcopy` here

File: roiextractors/segmentationextractor.py [hunk 5]
```diff
@@ -215,25 +154,66 @@ def get_image_size(self) -> ArrayType:
...
+    def get_images_dict(self):
+        """
+        Returns traces as a dictionary with key as the name of the ROiResponseSeries
+        Returns
+        -------
+        _roi_response_dict: dict
+            dictionary with key, values representing different types of Images used in segmentation:
+            Mean, Correlation image
+        """
+        return deepcopy(dict(mean=self._image_mean,
```
Comment (bendichter) [inline roiextractors/segmentationextractor.py:199]: I don't think you need a `deepcopy` here

File: roiextractors/segmentationextractor.py [hunk 6]
```diff
@@ -215,25 +154,66 @@ def get_image_size(self) -> ArrayType:
...
+                             correlation=self._image_correlation))
+
+    def get_images(self, name='correlation'):
```
Comment (bendichter) [inline roiextractors/segmentationextractor.py:202]: shouldn't this be `get_image`?

File: roiextractors/extractors/schnitzerextractor/cnmfesegmentationextractor.py (hunk 2 above)
Comment (Saksham20) [reply roiextractors/extractors/schnitzerextractor/cnmfesegmentationextractor.py:56]: i'm calling this summary image in all classes, but its assigned to `self._image_correlation` in the `__init__`

File: roiextractors/extractors/suite2p/suite2psegmentationextractor.py (hunk 3 above)
Comment (bendichter) [reply roiextractors/extractors/suite2p/suite2psegmentationextractor.py:63]: are you trying to change the data type? Or flip the image?

File: roiextractors/segmentationextractor.py [hunk 7]
```diff
@@ -3,6 +3,8 @@
 import numpy as np
 from .extraction_tools import ArrayType
 from .extraction_tools import _pixel_mask_extractor
+from copy import deepcopy
+import warnings
```
Comment (bendichter) [inline roiextractors/segmentationextractor.py:7]:
```suggestion
```

File: roiextractors/extractors/suite2p/suite2psegmentationextractor.py (hunk 3 above)
Comment (Saksham20) [reply roiextractors/extractors/suite2p/suite2psegmentationextractor.py:63]: this is how its implemented in the suite2p nwbwrite method: they have selected a subframe within the self.ops['Vcorr'] image to assign to the correlation image. This subset is defined by the keys 'xrange' and 'yrange' in self.ops. 
This is their implementation: https://github.com/MouseLand/suite2p/blob/main/suite2p/io/nwb.py#L272

File: roiextractors/segmentationextractor.py (hunk 5 above)
Comment (Saksham20) [reply roiextractors/segmentationextractor.py:199]: it does seem to make a difference: i tested it by manually changing the returned image_array (`get_images_dict()['mean']`) (without deepcopy) and it also changed the internal attribute (`self._image_mean`). With deepcopy this override does not alter the internal attribute.

File: roiextractors/segmentationextractor.py (hunk 7 above)
Comment (Saksham20) [reply roiextractors/segmentationextractor.py]: https://github.com/catalystneuro/roiextractors/pull/45/commits/f2ae79b86c9060c772720451349a96ff9eb6b0ba

File: roiextractors/segmentationextractor.py (hunk 6 above)
Comment (Saksham20) [reply roiextractors/segmentationextractor.py]: yes thanks for catching that, changed https://github.com/catalystneuro/roiextractors/pull/45/commits/f2ae79b86c9060c772720451349a96ff9eb6b0ba

File: roiextractors/extractors/suite2p/suite2psegmentationextractor.py (hunk 3 above)
Comment (bendichter) [reply roiextractors/extractors/suite2p/suite2psegmentationextractor.py:63]: oh, I see. OK, I think they were storing images from multiple planes side-by-side and then pulling the pixels corresponding to the plane of interest

**Pull request merged**